molecule 2.22

fxcoudert opened the pull request:
Move to Python 3: https://github.com/Homebrew/homebrew-core/issues/47050

bayandin commented on Formula/molecule.rb:
```diff
@@ -154,98 +163,153 @@ class Molecule < Formula
+  resource "molecule" do
+    url "https://files.pythonhosted.org/packages/61/75/1c78ba2d7299e87e42f62d55468245ecda4ac638bd9ae28d2c97a4a4796e/molecule-2.22.tar.gz"
+    sha256 "d9d7621167041ae2a8eb19f1f8dc23c071cdab2cd3ca80655e2c8796b4c00e09"
+  end
```
Is it expected that we have a resource coinciding with the formula?

fxcoudert replied:
Oops

fxcoudert commented:
```
Processing /private/tmp/molecule--ansible-lint-20191122-30113-10b1r96/ansible-lint-4.1.0
  Created temporary directory: /private/tmp/pip-req-build-ifqr1zsx
  Added file:///private/tmp/molecule--ansible-lint-20191122-30113-10b1r96/ansible-lint-4.1.0 to build tracker '/private/tmp/pip-req-tracker-d3djn4wz'
    Running setup.py (path:/private/tmp/pip-req-build-ifqr1zsx/setup.py) egg_info for package from file:///private/tmp/molecule--ansible-lint-20191122-30113-10b1r96/ansible-lint-4.1.0
    Running command python setup.py egg_info
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/tmp/pip-req-build-ifqr1zsx/setup.py", line 150, in <module>
        __name__ == '__main__' and setuptools.setup(**setup_params)
      File "/usr/local/Cellar/molecule/2.22/libexec/lib/python3.7/site-packages/setuptools/__init__.py", line 145, in setup
        return distutils.core.setup(**attrs)
      File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/core.py", line 108, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/local/Cellar/molecule/2.22/libexec/lib/python3.7/site-packages/setuptools/dist.py", line 447, in __init__
        k: v for k, v in attrs.items()
      File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/dist.py", line 292, in __init__
        self.finalize_options()
      File "/usr/local/Cellar/molecule/2.22/libexec/lib/python3.7/site-packages/setuptools/dist.py", line 735, in finalize_options
        ep.load()(self, ep.name, value)
      File "/usr/local/Cellar/molecule/2.22/libexec/lib/python3.7/site-packages/setuptools/dist.py", line 291, in check_specifier
        packaging.specifiers.SpecifierSet(value)
      File "/usr/local/Cellar/molecule/2.22/libexec/lib/python3.7/site-packages/setuptools/_vendor/packaging/specifiers.py", line 594, in __init__
        specifiers = [s.strip() for s in specifiers.split(",") if s.strip()]
    AttributeError: 'SpecifierSet' object has no attribute 'split'
```

fxcoudert commented:
Almost the same error, but in a different place now: 

```
  Getting requirements to build wheel: finished with status 'done'
    Created temporary directory: /private/tmp/pip-modern-metadata-01i4mf3_
    Preparing wheel metadata: started
    Running command /usr/local/Cellar/molecule/2.22/libexec/bin/python3.7 /usr/local/Cellar/molecule/2.22/libexec/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /private/tmp/tmp0x0qaovj
    Traceback (most recent call last):
      File "/usr/local/Cellar/molecule/2.22/libexec/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py", line 257, in <module>
        main()
      File "/usr/local/Cellar/molecule/2.22/libexec/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py", line 240, in main
        json_out['return_val'] = hook(**hook_input['kwargs'])
      File "/usr/local/Cellar/molecule/2.22/libexec/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py", line 110, in prepare_metadata_for_build_wheel
        return hook(metadata_directory, config_settings)
      File "/private/tmp/pip-build-env-093mj4nw/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 156, in prepare_metadata_for_build_wheel
        self.run_setup()
      File "/private/tmp/pip-build-env-093mj4nw/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 142, in run_setup
        exec(compile(code, __file__, 'exec'), locals())
      File "setup.py", line 350, in <module>
        __name__ == '__main__' and setuptools.setup(**setup_params)
      File "/private/tmp/pip-build-env-093mj4nw/overlay/lib/python3.7/site-packages/setuptools/__init__.py", line 145, in setup
        return distutils.core.setup(**attrs)
      File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/core.py", line 108, in setup
        _setup_distribution = dist = klass(attrs)
      File "/private/tmp/pip-build-env-093mj4nw/overlay/lib/python3.7/site-packages/setuptools/dist.py", line 447, in __init__
        k: v for k, v in attrs.items()
      File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/dist.py", line 292, in __init__
        self.finalize_options()
      File "/private/tmp/pip-build-env-093mj4nw/overlay/lib/python3.7/site-packages/setuptools/dist.py", line 735, in finalize_options
        ep.load()(self, ep.name, value)
      File "/private/tmp/pip-build-env-093mj4nw/overlay/lib/python3.7/site-packages/setuptools/dist.py", line 291, in check_specifier
        packaging.specifiers.SpecifierSet(value)
      File "/private/tmp/pip-build-env-093mj4nw/overlay/lib/python3.7/site-packages/setuptools/_vendor/packaging/specifiers.py", line 594, in __init__
        specifiers = [s.strip() for s in specifiers.split(",") if s.strip()]
    AttributeError: 'SpecifierSet' object has no attribute 'split'
    Preparing wheel metadata: finished with status 'error'
Cleaning up...
  Removing source in /private/tmp/pip-req-build-ywemclw9
Removed file:///private/tmp/molecule-20191122-59791-2i9i2k/molecule-2.22 from build tracker '/private/tmp/pip-req-tracker-u9z66vrr'
Removed build tracker '/private/tmp/pip-req-tracker-u9z66vrr'
ERROR: Command errored out with exit status 1: /usr/local/Cellar/molecule/2.22/libexec/bin/python3.7 /usr/local/Cellar/molecule/2.22/libexec/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /private/tmp/tmp0x0qaovj Check the logs for full command output.
```

Bo98 commented:
Should be fixed by the setup.py changes in https://github.com/ansible/molecule/commit/c9fee498646a702c77b5aecf6497cff324acd056
